### PR TITLE
ortho proj constructor

### PR DESCRIPTION
**Pull request by abuts**
this should address #1005. The sample provided in the ticket works, despite its interface is outdate. 
The description modified according to the new reality

### CODE REVIEW

**Review by cmarooney-stfc — CHANGES_REQUESTED** [review]

File: horace_core/sqw/coord_transform/@aProjectionBase/aProjectionBase.m [hunk 1]
```diff
@@ -93,7 +93,7 @@
         % returns true if lattice angles have been set up
         angdeg_defined
         % old interface to img_offset for old scripts accepting a
-        % structure, containign this value
+        % structure, containign this value or old user scripts
```
[the remark this answers is not in the document]
Comment (abuts) [reply horace_core/sqw/coord_transform/@aProjectionBase/aProjectionBase.m:96]: hopefylly clarified -- old user scripts defining a structure with this value. The description modified

File: horace_core/sqw/coord_transform/@aProjectionBase/aProjectionBase.m [hunk 2]
```diff
@@ -65,12 +65,12 @@
         % particular projection-projection pair of transformations,
         % optimized for specific projection-projection pair of classes
         do_generic;
-        % testing property. Normaly transformation from source to target
+        % testing property. Normally transformation from source to target
         % coordinate system in cut can be optimized as each transformation
         % is described by transformation matrices and the final
         % transformation is the production of all these matrices.
```
Comment (oerc0122) [inline horace_core/sqw/coord_transform/@aProjectionBase/aProjectionBase.m:71]: Product

File: horace_core/sqw/coord_transform/@aProjectionBase/aProjectionBase.m [hunk 3]
```diff
@@ -128,7 +129,7 @@
         % specific pairs of the projection types if such optimization
         % is available
         do_generic_ = true;
-        % if true, disables optimization of the transfornation from source
+        % if true, disables optimization of the transformation from source
         % to target coordinate system.
         disable_srce_to_targ_optimization_ = false;
```
Comment (oerc0122) [inline horace_core/sqw/coord_transform/@aProjectionBase/aProjectionBase.m:134]: Try to keep consistent abbreviations. Everywhere else uses `source` in full

File: horace_core/sqw/coord_transform/@ortho_proj/ortho_proj.m [hunk 4]
```diff
@@ -37,12 +40,14 @@
     %
     %IMPORTANT:
     % if you want to use ortho_proj as input for cut algorithm, it needs
-    % minimum two input parameters u and v, as lattice parameters for cut
-    % will be taken from sqw object if not provided with projection.
+    % minimum two input parameters u and v, (or their default values) as
```
Comment (oerc0122) [inline horace_core/sqw/coord_transform/@ortho_proj/ortho_proj.m:43]:
```suggestion
    % a minimum of two input parameters u and v, (or their default values) as
```
or
```suggestion
    % at least two input parameters u and v, (or their default values) as
```

File: horace_core/sqw/coord_transform/@ortho_proj/ortho_proj.m [hunk 5]
```diff
@@ -37,12 +40,14 @@
...
+    % the lattice parameters for cut will be taken from sqw object
+    % if not provided with projection.
     %
     % For independent usage u,v and lattice parameters (minimal functional
     % form) needs to be specified. Any other parameters have their reasonable
```
Comment (oerc0122) [inline horace_core/sqw/coord_transform/@ortho_proj/ortho_proj.m]: Not sure what
> minimal functional form

means here.
Clarify

File: horace_core/sqw/coord_transform/@ortho_proj/ortho_proj.m [hunk 6]
```diff
@@ -585,7 +590,7 @@
             % suggesting the reason for failure if the inputs are incorrect
             % w.r.t. each other.
             %
-            % Set's up the transformation caches
+            % Set's up the internal  image transformation caches.
```
Comment (oerc0122) [inline horace_core/sqw/coord_transform/@ortho_proj/ortho_proj.m:593]:
```suggestion
            % Sets up the internal image transformation caches.
```

File: horace_core/sqw/coord_transform/@ortho_proj/ortho_proj.m (hunk 5 above)
Comment (abuts) [reply horace_core/sqw/coord_transform/@ortho_proj/ortho_proj.m]: its defined in the previous sentence.

File: horace_core/sqw/coord_transform/@aProjectionBase/aProjectionBase.m [hunk 7]
```diff
@@ -92,8 +92,9 @@
         alatt_defined
         % returns true if lattice angles have been set up
         angdeg_defined
-        % old interface to img_offset for old scripts accepting a
-        % structure, containign this value
+        % old interface to img_offset for old data containing a
+        % structure with this value or old user scripts, which define
```
Comment (cmarooney-stfc) [inline horace_core/sqw/coord_transform/@aProjectionBase/aProjectionBase.m:96]:
```suggestion
        % structure with this value, or old user scripts which define
```
I've fixed the punctuation, but "this value" still hasn't been defined. Presumably old data had a structure with uoffset in it and now you have a property to get/set it but you need to define what uoffset is and how it relates to img_offset

File: horace_core/sqw/coord_transform/@aProjectionBase/aProjectionBase.m [hunk 8]
```diff
@@ -297,13 +298,14 @@
             % expressed in Crystal Cartesian coordinate system with a
             % vector in the coordinate system attached to reciprocal lattice.
             % Optional Input:
-            % ndim -- if provided and equal to 4, retun the 4x4 matrix
+            % ndim -- if provided and equal to 4, return the 4x4 matrix
             %         rather then 3x3 standard matrix, with unit expansion
-            %         to forth dimension (e.g. add rows/columnth with 0
+            %         to forth dimension (e.g. add rows/columns with 0
```
Comment (cmarooney-stfc) [inline horace_core/sqw/coord_transform/@aProjectionBase/aProjectionBase.m:303]:
```suggestion
            %         to fourth dimension (e.g. add rows/columns with 0
```

File: horace_core/sqw/coord_transform/@ortho_proj/ortho_proj.m [hunk 9]
```diff
@@ -11,13 +11,21 @@
     %   >> proj = ortho_proj(proj_struct)
     %             where proj_struct is the
     %             structure, containing any fields, with names, equal any
-    %             public fields of the ortho_proj class
+    %             public fields of the ortho_proj class.
+    %
+    % As a standard serializable class accepts full set of positional and
```
Comment (cmarooney-stfc) [inline horace_core/sqw/coord_transform/@ortho_proj/ortho_proj.m:16]:
```suggestion
    % As a standard serializable class, class ortho_proj accepts full set of positional and
```
I'm adding text to make this more meaningful but I might have guessed the meaning wrong

File: horace_core/sqw/coord_transform/@ortho_proj/ortho_proj.m [hunk 10]
```diff
@@ -11,13 +11,21 @@
...
+    % key-value parameters, which constitute its properties
```
Comment (cmarooney-stfc) [inline horace_core/sqw/coord_transform/@ortho_proj/ortho_proj.m:17]: I'm not clear why you need to see this. Surely this is true for all serializable classes. What particular information are you pointing to with this generalisation.

File: horace_core/sqw/coord_transform/@ortho_proj/ortho_proj.m [hunk 11]
```diff
@@ -27,6 +35,19 @@
     %                   :
     %   >> proj = ortho_proj(...,'lab4',labelstr,...)
     %
+    % Minimal functional form:
+    %   >> proj =  ortho_proj(u,v,'alatt',lat_param,'angdeg',lattice_angles_in_degrees);
+    %
+    %IMPORTANT:
+    % if you want to use ortho_proj as input for cut algorithm, it needs
```
Comment (cmarooney-stfc) [inline horace_core/sqw/coord_transform/@ortho_proj/ortho_proj.m:42]:
```suggestion
    % if you want to use ortho_proj as input for the cut algorithm, it needs
```

File: horace_core/sqw/coord_transform/@ortho_proj/private/init_by_input_parameters_.m [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+%outdate forms of input of the ortho_proj constructor.
+%
+% Inputs:
+% list of input parameters of projection, produced as set of positional
+% parameters and then key-value, key-value pairs.
+%
+% The positional parameters order is:
+%
+% u,v,w,nonorthogonal,type,alatt,angdeg,offset,label,title,lab1,lab2,lab3,lab4
+% First non-positional parameter considered to be key.
```
Comment (cmarooney-stfc) [inline horace_core/sqw/coord_transform/@ortho_proj/private/init_by_input_parameters_.m:12]: It's not clear how to use this. What are the key parameters? Or can the positional parameters also be key parameters. Some more explanation is needed

**Review by cmarooney-stfc — CHANGES_REQUESTED** [review]

File: horace_core/sqw/coord_transform/@aProjectionBase/aProjectionBase.m (hunk 7 above)
Comment (abuts) [reply horace_core/sqw/coord_transform/@aProjectionBase/aProjectionBase.m:96]: the property value?

File: horace_core/sqw/coord_transform/@ortho_proj/ortho_proj.m (hunk 10 above)
Comment (abuts) [reply horace_core/sqw/coord_transform/@ortho_proj/ortho_proj.m:17]: I am reminding the obvious and in the next sentence proivde the exact form of the constructor

File: horace_core/sqw/coord_transform/@aProjectionBase/aProjectionBase.m (hunk 7 above)
Comment (abuts) [reply horace_core/sqw/coord_transform/@aProjectionBase/aProjectionBase.m:96]: we are talking about the property and its value

File: horace_core/sqw/coord_transform/@ortho_proj/ortho_proj.m [hunk 13]
```diff
@@ -70,9 +91,9 @@
     % Also accepts these and aProjectionBase properties as set of key-values
     % pairs following standard serializable class constructor agreements.
     %
-
-    % Original author: T.G.Perring
-    %
+    % NOTE:
+    % constructor does not accept legacy ub_inv_legacy matrix, specified in
```
[the remark this answers is not in the document]
Comment (abuts) [reply horace_core/sqw/coord_transform/@ortho_proj/ortho_proj.m:95]: if you want this form, then even if would be more appropriate.

File: horace_core/sqw/coord_transform/@ortho_proj/private/init_by_input_parameters_.m (hunk 12 above)
Comment (abuts) [reply horace_core/sqw/coord_transform/@ortho_proj/private/init_by_input_parameters_.m:12]: this is standard usage of serializable. You can not say the same thing for each of its instances.

File: horace_core/sqw/coord_transform/@ortho_proj/private/init_by_input_parameters_.m [hunk 14]
```diff
@@ -0,0 +1,50 @@
...
+
+% constructor does not accept legacy alignment matrix
+opt =  [ortho_proj.fields_to_save_(1:end-1);aProjectionBase.init_params(:)];
+% check if the type is defined explicitly
+n_type = find(ismember(opt,'type'));
+text_in = cellfun(@(x)char(string(x)),varargin,'UniformOutput',false); 
+
+if ismember('type',text_in) || ... % defined as key-value pair
```
[the remark this answers is not in the document]
Comment (abuts) [reply horace_core/sqw/coord_transform/@ortho_proj/private/init_by_input_parameters_.m:20]: standard usage of ismember -- returns array of logical outputs with size equal to the size of the first parameters. In both cases the requests are different.

File: horace_core/sqw/file_io/@sqw_binfile_common/get_sqw.m [hunk 15]
```diff
@@ -111,7 +111,7 @@
 % by alignment matrix
 header_av = exp_info.header_average;
 u_to_rlu  = header_av.u_to_rlu(1:3,1:3);
-if any(abs(lower_part(u_to_rlu))>1.e-7) % if all 0, its B-matrix so certainly
+if any(abs(lower_part(u_to_rlu))>1.e-7) % if all 0, its inverted B-matrix so certainly
```
[the remark this answers is not in the document]
Comment (abuts) [reply horace_core/sqw/file_io/@sqw_binfile_common/get_sqw.m:114]: two rows comment here. With two rows looks clear to me

**Pull request merged**